spec/javascripts/controllers/middleware/middleware_topology_controller_spec.js

```diff
@@ -65,28 +73,42 @@ describe('middlewareTopologyController', function () {
     });
   });
 
+
   describe('kinds contains all expected mw kinds', function () {
     it('in all main objects', function () {
-      expect(Object.keys(scope.kinds).length).toBe(3);
+      expect(Object.keys(scope.kinds).length).toBe(4);
       expect(scope.kinds["MiddlewareManager"]).toBeDefined();
       expect(scope.kinds["MiddlewareServer"]).toBeDefined();
       expect(scope.kinds["MiddlewareDeployment"]).toBeDefined();
+      expect(scope.kinds["MiddlewareDatasource"]).toBeDefined();
     });
   });
 
+
   describe('the mw topology gets correct icons', function () {
     it('in graph elements', function () {
       expect($controller.class_name(mw_manager)).toContain("vendor-hawkular");
       expect($controller.class_name(mw_server)).toContain("vendor-hawkular");
       expect($controller.class_name(mw_deployment)).toContain("middleware_deployment_war");
+      expect($controller.class_name(mw_datasource)).toBeUndefined()
```

Comment:
`class_name()`:

```javascript
      self.class_name = function class_name(d) {
          var class_name = d.item.icon;
          return class_name;
      }
```

For datasources the icon is not set, because it's done via the `foo.kind_of?(MiddlewareDatasource)` which is not perfect, I'll change that to use the font icon from the decorator.